gatorgrouper/utils/group_creation.py

```diff
@@ -1,43 +1,85 @@
+) -> List[List[str]]:
+    """ Calculate number of groups based on desired students per group """
+    # number of groups = number of students / minimum students per group
+    numgrp = int(len(responses) / grpsize)
+
+    return group_random_num_group(responses, numgrp, conflicts)
+
+
+# pylint: disable=dangerous-default-value
+# pylint: disable=too-many-locals
+def group_random_num_group(
+    responses: str, numgrp: int, conflicts=[("foo", "bar", -1)]
+) -> List[List[str]]:
     """ group responses using randomization approach """
-    # number of students placed into a group
-    stunum = 0
-    iterable = iter(responses)
-    # number of students in each group (without overflow)
-    grpsize = int(len(responses) / numgrp)
-    groups = list()
-    for _ in range(0, numgrp):
-        group = list()
-        while len(group) is not grpsize and stunum < len(responses):
-            group.append(next(iterable))
+    intensity = 100
+    # Intensity is the value that represents the number of attempts made to group
+    optimized_groups = list()
+    # Optimized Groups holds the groups after scoring maximization
+    top_ave = -10000000
+    # Top Average is our check to see if the group made is better than the group we have
+    while intensity > 0:
+        # number of students placed into a group
+        stunum = 0
+        iterable = iter(responses)
+        # number of students in each group (without overflow)
+        grpsize = int(len(responses) / numgrp)
+        groups = list()
+        for _ in range(0, numgrp):
+            group = list()
+            while len(group) is not grpsize and stunum < len(responses):
+                group.append(next(iterable))
+                stunum = stunum + 1
+            groups.append(group)
+        # deal with the last remaining students
+        if len(responses) % stunum != 0:
+            logging.info("Overflow students identified; distributing into groups.")
+        for _x in range(0, len(responses) % stunum):
+            groups[_x].append(next(iterable))
             stunum = stunum + 1
-        groups.append(group)
-    # deal with the last remaining students
-    if len(responses) % stunum != 0:
-        logging.info("Overflow students identified; distributing into groups.")
-    for _x in range(0, len(responses) % stunum):
-        groups[_x].append(next(iterable))
-        stunum = stunum + 1
-
-    # scoring and return
-    scores, ave = [], 0
-    scores, ave = group_scoring.calculate_avg(groups)
-    logging.info("scores: %s", str(scores))
-    logging.info("average: %d", ave)
-    return groups
+        # scoring and return
+        # a list of the conflict scores to affect group scores
+        conflict_scores = []
+        for grp in groups:
+            # iterate through groups
+            for confs in conflicts:
+                # iterate through conflicts given as args
+                if (confs[0] or confs[1]) in grp:
```

Comment:
This does not make logical sense, and tests for whether a boolean is in the group.

Reply:
`confs[0]` and `confs[1]` are strings, not booleans, as given in the format of the argument of the function. `grp` is a list of strings.

Reply:
But `(confs[0] or confs[1])` results in a boolean. Then (order of precedence) you compare that boolean to see if it is `in grp`.

Reply:
Ah, I see now. Fixed this issue in recent commit.

Reply:
I'm not sure if it totally is -- does this logic mean that if either of conflict participants are in a group, the group is scored worse? Should it not be if both of them are (thus an actual conflict is produced)?

Reply:
You're correct, the `or` should have been an `and`, missed that when separating the conditions.